remotes/docker/httpreadseeker.go

```diff
@@ -54,7 +58,22 @@ func (hrs *httpReadSeeker) Read(p []byte) (n int, err error) {
 		return 0, err
 	}
 
-	n, err = rd.Read(p)
+	var schedC <-chan time.Time
+
+	schedC = time.After(readerTimeout * time.Second)
```

Comment:
I don't think that we should use fixed timeout number here.

Currently, we already have traffic checker for CRI pull image protocol. If there is no traffic in one minute, it will abort the request. If you are using containerd client API instead of CRI, please consider to extract the functionality from CRI.